libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -170,23 +175,38 @@ void TLSSocket::closeStream() {
  * @param listeners number of listeners in the queue
  */
 TLSSocket::TLSSocket(const std::shared_ptr<TLSContext> &context, const std::string &hostname, const uint16_t port, const uint16_t listeners)
-    : Socket(context, hostname, port, listeners),
-      ssl_(0) {
+    : Socket(context, hostname, port, listeners) {
   logger_ = logging::LoggerFactory<TLSSocket>::getLogger();
   context_ = context;
 }
 
 TLSSocket::TLSSocket(const std::shared_ptr<TLSContext> &context, const std::string &hostname, const uint16_t port)
-    : Socket(context, hostname, port, 0),
-      ssl_(0) {
+    : Socket(context, hostname, port, 0) {
   logger_ = logging::LoggerFactory<TLSSocket>::getLogger();
   context_ = context;
 }
 
-TLSSocket::TLSSocket(const TLSSocket &&d)
-    : Socket(std::move(d)),
-      ssl_(0) {
-  context_ = d.context_;
+TLSSocket::TLSSocket(TLSSocket &&other) noexcept
+    : Socket(std::move(other)),
+      context_{ utils::exchange(other.context_, nullptr) } {
+  std::lock_guard<std::mutex> lg{ other.ssl_mutex_ };
```

Comment:
std::mutex::lock() (and, consequently std::lock_guard::lock_guard()) can throw, so neither this, nor the move assignment operator can be `noexcept`.